Ignore all cursor management requests while in VR mode:

hai-vr opened the pull request:
- When entering VR mode, unlock cursor and ignore all cursor management requests.
- This new behaviour prevents the cursor from being stolen from other VR desktop overlay applications.

hai-vr committed:
Ignore all cursor management requests while in VR mode:
- When entering VR mode, unlock cursor and ignore all cursor management requests.
- This new behaviour prevents the cursor from being stolen from other VR desktop overlay applications.

dooly123 commented:
Heyo the bool should be swapped for my is in vr check that way it's one spot not two. Should be a helper function somewhere

hai-vr committed:
In cursor management request checks, use BasisDeviceManagement.IsUserInDesktop():
- Switch to BasisDeviceManagement.IsUserInDesktop() for checking cursor management requests.
- Bypass checks to unlock the cursor when switching from Desktop to VR.

hai-vr commented:
Updated to use `BasisDeviceManagement.IsUserInDesktop()`
